src/Microsoft.PowerApps.TestEngine.Tests/Reporting/TestLoggerTests.cs

```diff
@@ -57,10 +59,24 @@ public void IsEnabledTest(LogLevel level, bool shouldBeEnabled)
         [Fact]
         public void WriteToLogsFileThrowsOnInvalidPathTest()
         {
-            MockFileSystem.Setup(x => x.IsValidFilePath(It.IsAny<string>())).Returns(false);
             var testLogger = new TestLogger(MockFileSystem.Object);
-            Assert.Throws<ArgumentException>(() => testLogger.WriteToLogsFile("", ""));
-            MockFileSystem.Verify(x => x.IsValidFilePath(""), Times.Once());
+            var createdLogs = new Dictionary<string, string[]>();
+
+            MockFileSystem.Setup(x => x.CreateDirectory(It.IsAny<string>()));
+            MockFileSystem.Setup(x => x.Exists(It.IsAny<string>())).Returns(false);
+            MockFileSystem.Setup(x => x.IsValidFilePath(It.IsAny<string>())).Returns(false);
+            MockFileSystem.Setup(x => x.WriteTextToFile(It.IsAny<string>(), It.IsAny<string[]>())).Callback((string filePath, string[] logs) =>
+            {
+                createdLogs.Add(filePath, logs);
+            });
+
+            var stringWriter = new StringWriter();
+            Console.SetOut(stringWriter);
+
+            testLogger.WriteToLogsFile("C:\\invalidPath", "");
+
+            var output = stringWriter.ToString();
+            Assert.Contains("[Critical Error]: Could not find a part of the path", output);
```

Comment:
I do not see any critical error logged onto console within  testLogger.WriteToLogsFile() method.

Reply:
I had suggested that, as at this point in the execution sequence, it is safe to have the built-in exception handler for `CreateDirectory` handle this - this would ONLY happen when the directory cannot be created due to external reasons (disk busy, I/O error, etc). I'm not too particular about any particular way at this point, it would be more about how we phrase this error to the users, vs just printing the built-in exception's message string.

Reply:
I am not saying we need that exception or error to be thrown. I meant this unit test is failing for that reason 
``` Assert.Contains("[Critical Error]: Could not find a part of the path", output);``` is going to fail